specs/dynamicdependencies/DynamicDependencies.md

```diff
@@ -0,0 +1,1366 @@
+// (package family name, minimum version, etc)
+// may match multiple packages, but ensures Deployment won't remove
+// a package if it's the only one satisfying the PackageDependency.
+//
+// @note A package matching a PackageDependency pin can still be removed
+//       as long as there's another that satisfies the PackageDependency.
+//       For example, if Fwk-v1 is installed and a PackageDependency specifies
+//       MinVersion=1 and then Fwk-v2 is installed, Deployment could remove
+//       Fwk-v1 because Fwk-v2 will satisfy the PackageDependency. After Fwk-v1
+//       is removed Deployment won't remove Fwk-v2 because it's the only package
+//       satisfying the PackageDependency. Thus  Fwk-v1 and Fwk-v2 (and any other
+//       package matching the PackageDependency) are 'loosely pinned' â€“ Deployment
+//       guarantees it won't remove a package if it would make a PackageDependency
+//       unsatisfied.
+//
+// A PackageDependency specifies criteria (package family, minimum version, etc)
+// and not a specific package. Deployment reserves the right to use a different
+// package (e.g. higher version) to satisfy the PackageDependency if/when
+// one becomes available.
+//
+// @param user the user scope of the package dependency. If NULL the caller's
+//        user context is used. MUST be NULL if MddPinPackageDependency::ScopeIsSystem
+//        is specified
+// @param lifetimeArtifact MUST be NULL if MddMddPinPackageDependency::LifecycleHint_Process (default)
+// @param packageDependencyId allocated via LocalAlloc; use LocalFree to deallocate
+//
+// @note MddPinPackageDependency() fails if the PackageDependency cannot be resolved to a specific
+//       package. This package resolution check is skipped if
+//       MddPinPackageDependency::DoNotVerifyDependencyResolution is specified. This is useful
+//       for installers running as user contexts other than the target user (e.g. installers
+//       running as LocalSystem).
+STDAPI MddPinPackageDependency(
+    PSID user,
+    _In_ PCWSTR packageFamilyName,
+    PACKAGE_VERSION minVersion,
+    MddPackageDependencyProcessorArchitectures packageDependencyProcessorArchitectures,
+    PCWSTR lifetimeArtifact,
+    MddPinPackageDependency flags,
+    _Outptr_result_maybenull_ PWSTR* packageDependencyId);
+
+// Undefine a package dependency. Removing a pin on a PackageDependency is typically done at uninstall-time.
+// This implicitly occurs if the package dependency's 'lifetime artifact' (specified via MddPinPackageDependency)
+// is deleted. Packages that are not referenced by other packages and have no pins are elegible to be removed.
+//
+// @warn MddUnpinPackageDependency() requires the caller have administrative privileges
+//       if the package dependency was pinned with MddPinPackageDependency::ScopeIsSystem.
+STDAPI MddUnpinPackageDependency(
+    _In_ PCWSTR packageDependencyId);
+
+// Resolve a previously-pinned PackageDependency to a specific package and
```

Comment:
All this delicious flavor text should probably go in the API description part, rather than here as a comment?

Reply:
Typo. Those should be doccomments ///